system-addon/content-src/components/ContextMenu/ContextMenu.jsx

```diff
@@ -24,9 +24,23 @@ class ContextMenu extends React.Component {
   componentDidUnmount() {
     window.removeEventListener("click", this.hideContext);
   }
-  onKeyDown(index) {
-    if (index > this.props.tabbableOptionsLength) {
-      this.hideContext();
+  onKeyDown(event, option) {
+    switch (event.key) {
+      case "Tab":
+        if (event.shiftKey) {
+          if (option.first) {
+            this.hideContext();
+          }
+          return;
+        }
+        if (option.last) {
+          this.hideContext();
```

Comment:
Should be able to combine this check and body with above:
```js
if (event.shiftKey && option.first || option.last) {
  this.hideContext()
```
With a comment describing that this is to handle tabbing past the front or end